### ISSUE TITLE
[BUG] 3006rc repo keyring doesn't have current key

### ISSUE THREAD

**Issue by OrangeDog** [saltstack/salt#63925]
**Description**
A new GPG key has been created for the 3006 RCs.
However, there is no keyring available that contains both the current and new keys.

**Steps to Reproduce the behavior**
Follow the [RC install instructions](https://docs.saltproject.io/salt/install-guide/en/latest/topics/release-candidate.html), which includes
```bash
sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring.gpg https://staging.repo.saltproject.io/salt_rc/salt/py3/ubuntu/22.04/amd64/minor/3006.0rc1/SALT-PROJECT-GPG-PUBKEY-2023.gpg
```
This overwrites the current Salt keyring with the new one. As the new one doesn't contain the old key, all current repos then fail to verify.

**Expected behavior**
The latest keyring should include all current valid keys.

**Comment by dmurphy18** [comment]
@OrangeDog  Also note that the instructions for the Release Candidate need correcting for rc2 and not leaving existing entries in the salt.list file.  Created documentation issue for this https://github.com/saltstack/salt/issues/63953

Close this if it addresses your concerns with the doc issue.

**Comment by OrangeDog** [comment]
No, it's a repo management issue. If you're going to change the keys you either need to change all of them, or you need to provide a keyring with all the keys in it.

The instructions would be correct if the repo was set up correctly.

**Comment by dmurphy18** [comment]
@OrangeDog The RC area needs cleanup, the 3005rc1 and 3005rc2 areas should have been gone long ago, and other areas too.  I checked where the current RC files are placed and they look good, see https://staging.repo.saltproject.io/salt_rc/salt/py3/redhat/9/x86_64/minor/3006.0rc2/

The keys are changed in all of the minion locations, it is just the old cruft that is confusing and I have requested it be removed, but that will probably occur after release at this point.

**Comment by OrangeDog** [comment]
Perhaps I should be clearer.

This keyring: https://staging.repo.saltproject.io/salt_rc/salt/py3/ubuntu/22.04/amd64/minor/3006.0rc1/SALT-PROJECT-GPG-PUBKEY-2023.gpg
Does not contain the signing key for e.g. this repo: https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3004

Either every existing repo (including all the archived ones) should be updated to use the new key, or the new keyring should include the old key.

**Comment by dmurphy18** [comment]
@OrangeDog  Salt is done in separate repo's and they are left as-is at time of creation and release, hence they are treated as read-only, so older releases are not touched, and to update a Salt version would require a new release for that version, that is, frozen in time.

So what you want will not happen.  As for Salt 3004, support for it ends 18 April 2023, hence this could only apply for Salt major version 3005, but with the switch a new build/release pipeline using GitHub Actions for 3006, building for Salt major version 3005 would be mostly due to a major CVE found.

Sorry but don't believe in going back and changing what was released at some previous point in time.
If something is to be updated, then it should be in a new version.

Closing this since the issue is really documentation for the Release Candidate and that is being addressed in the issue above, including the erroneous ```tee --append``` which really is the problem in not overwriting an existing salt.list.

Please re-open if there is something else to consider apart from the issues already addressed

**Comment by OrangeDog** [comment]
What I want is for you to put the current key in the new keyring, as the original issue says. Why will you not do this?

**Comment by dmurphy18** [comment]
So you want a new keyring with both the old SHA-1 and SHA-256 keys in it ?

Looking for clarification.

Noting that SHA-1 keys are getting dropped and not supported with new OSs, and OpenSSL v3.x

**Comment by OrangeDog** [comment]
This is from the latest onedir repo:
```
$ gpg --show-keys SALTSTACK-GPG-KEY.pub
pub   rsa2048 2014-06-24 [SC]
      754A1A7AE731F165D5E6D4BD0E08A149DE57BFBE
uid                      SaltStack Packaging Team <packaging@saltstack.com>
sub   rsa2048 2014-06-24 [E]
```

And this is the new RC one:
```
$ gpg --show-keys SALT-PROJECT-GPG-PUBKEY-2023.gpg
pub   rsa3072 2023-02-01 [SC]
      10857FFDD3F91EAE577A21D664CBBC8173D76B3F
uid                      Salt Project Packaging <saltproject-packaging@vmware.com>
sub   rsa3072 2023-02-01 [E]
```

Both should be in the new keyring, so the current repos continue to work.
